middleware/beta.go

```diff
@@ -24,3 +25,23 @@ func BetaApiHandler(enableBetaRestriction bool, h http.Handler) http.Handler {
 		h.ServeHTTP(w, r)
 	})
 }
+
+func isBetaDomain(r *http.Request) bool {
+	return strings.HasPrefix(r.Host, "api.beta")
+}
+
+func isInternalTraffic(r *http.Request) bool {
+
+	// exclude the port from the potential IP address
+	host, _, err := net.SplitHostPort(r.Host)
+	if err != nil {
+		// if we fail to split from the port, just use the original host value
+		host = r.Host
+	}
+
+	return isValidIP(host)
```

Comment:
I think we may want to have a check here for `localhost` as well, just to be thorough.

```suggestion
	return isValidIP(host) || host == "localhost"
```